src/core/a-entity.js

```diff
@@ -568,11 +627,22 @@ function checkComponentDefined (el, name) {
   if (el.defaultComponents[name] !== undefined) { return true; }
 
   // Check if element contains the component.
-  if (el.hasAttribute(name)) { return true; }
+  if (el.components[name] && el.components[name].attrValue) { return true; }
 
   return isComponentMixedIn(name, el.mixinEls);
 }
 
+function getMixedInComponents (entityEl) {
+  var components = [];
+  entityEl.mixinEls.forEach(function (mixinEl) {
+    Object.keys(mixinEl.attrs).forEach(addComponent);
+    function addComponent (key) {
```

Comment:
OK. though i'd probably argue it improves readability there, but doesn't really help here.